demos/browser/app/meeting/CloudWatchLogger.ts

```diff
@@ -0,0 +1,134 @@
+ *   logger.debug(debugFunc()); // this will show up
+ *
+ * ```
+ */
+export default class CloudWatchLogger implements Logger {
+  name: string;
+  level: LogLevel;
+  logCapture: string[] = [];
+  meetingID: string;
+  attendeeID: string;
+  logSeqNo: any;
+
+  constructor(name: string, level = LogLevel.WARN, meetingID: string, attendeeID: string) {
+    this.name = name;
+    this.level = level;
+    this.meetingID = meetingID;
+    this.attendeeID = attendeeID;
+    this.logSeqNo = 0;
+  }
+
+  batchSizes = 50;
+  lock = false;
+  async publishToCloudWatch(base_url: string) {
+    var _this = this;
+
+    setInterval(async function () {
+
+      if (_this.lock == true || _this.logCapture.length == 0 )
+        return
+      _this.lock = true;
+      var batch = _this.logCapture.slice(0, _this.batchSizes);
+      const response = await fetch(
+        `${base_url}logs`, {
+          method: 'POST',
+          body: JSON.stringify({
+            logs: batch
+          })
+        }
+      );
+
+      const json = await response.json();
+      if (json.error) {
+        throw new Error(`Server error: ${json.error}`);
+      }
+      // delete elements upto current_size(logCapture) from the array logCapture
+      _this.logCapture = _this.logCapture.slice(batch.length);
+      _this.lock = false;
+    }, 5000);
+  }
+
+    info(msg: string): void {
+        this.log(LogLevel.INFO, msg);
+    }
+
+    warn(msg: string): void {
+        this.log(LogLevel.WARN, msg);
+    }
+
+    error(msg: string): void {
+        this.log(LogLevel.ERROR, msg);
+    }
+
+    debug(debugFunction: () => string): void {
+        if (LogLevel.DEBUG < this.level) {
+            return;
+        }
+        this.log(LogLevel.DEBUG, debugFunction());
+    }
+
+    setLogLevel(level: LogLevel): void {
+        this.level = level;
+    }
+
+    getLogLevel(): LogLevel {
+        return this.level;
+    }
+
+    private log(type: LogLevel, msg: string): void {
+        if (type < this.level) {
+            return;
+        }
+        const date = new Date();
+        const timestamp = date.toISOString();
+        const logMessage = `${timestamp} [${LogLevel[type]}] ${this.name} - ${msg}`;
+        //const logMessageCW = `${timestamp} MeetingId=${this.meetingID} AttendeeId=${this.attendeeID} [${LogLevel[type]}] ${this.name} - ${msg}`;
+
+        switch (type) {
+            case LogLevel.ERROR:
+                console.trace(logMessage);
+                break;
+            case LogLevel.WARN:
+                console.warn(logMessage);
+                break;
+            case LogLevel.DEBUG:
+                console.debug(logMessage.replace(/\\r\\n/g, '\n'));
+                break;
+            case LogLevel.INFO:
+                console.info(logMessage);
+                break;
+        }
+        var logJSON = {
+           "logSeqNo": this.logSeqNo,
+           "logMessage" : msg,
+           "timestamp" : date.getTime(),
+           "meetingID" : this.meetingID,
```

Comment:
I think you can send meetingID, attendeeID and appName in the POST request as they will be the same for every log message.